apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.GenerateClient/Program.cs

```diff
@@ -183,42 +240,52 @@ public override SyntaxNode VisitMethodDeclaration(MethodDeclarationSyntax node)
             {
                 try
                 {
+                    _method = _semanticModel.GetDeclaredSymbol(node);
                     _requestParameterSyntax = node.ParameterList.Parameters[0];
 
-                    var method = _semanticModel.GetDeclaredSymbol(node);
                     node = (MethodDeclarationSyntax)base.VisitMethodDeclaration(node);
 
-                    // If the method's return type is defined in the underlying client's class, qualify the return type.
-                    // However, if we've added a custom stream type in the main namespace and the name matches, skip the
-                    // qualification: we want it to point to the custom type.
-                    if (method.ReturnType.ContainingType == method.ContainingType &&
-                        !s_customStreams.Contains(method.ReturnType.Name))
+                    if (s_customStreamMethods.TryGetValue(_method.Name, out var customStreamMethodInfo) &&
+                        customStreamMethodInfo.TypeName != null)
+                    {
+                        node = node.WithReturnType(ParseTypeName(customStreamMethodInfo.TypeName));
+                    }
+                    else if (_method.ReturnType.ContainingType == _method.ContainingType)
                     {
+                        // If the method's return type is defined in the underlying client's class, qualify the return type.
                         node = node.WithReturnType(
                             QualifiedName(
-                                IdentifierName(method.ContainingType.Name),
-                                IdentifierName(method.ReturnType.Name)));
+                                IdentifierName(_method.ContainingType.Name),
+                                IdentifierName(_method.ReturnType.Name)));
                     }
 
                     return node;
                 }
                 finally
                 {
                     _requestParameterSyntax = null;
+                    _method = null;
                 }
             }
 
             public override SyntaxNode VisitXmlElement(XmlElementSyntax node)
             {
                 node = (XmlElementSyntax)base.VisitXmlElement(node);
 
-                // Append request param tags with a note about how the AppProfileId will be added to the request object
-                // if unspecified.
-                if (node.StartTag.Name.LocalName.ToString() == "param" &&
+                var content = node.Content;
+
+                if (s_customStreamMethods.TryGetValue(_method.Name, out var customStreamMethodInfo) &&
+                    customStreamMethodInfo.CustomDocs.TryGetValue(node.StartTag.ToString(), out var comment))
```

Comment:
Could we use the start tag name rather than the ToString representation? Then replace "<summary>" with "summary" in the dictionary.

Reply:
Done